### ISSUE TITLE
Help Wanted in refactoring to use vertex arrays

### ISSUE THREAD

**Issue by jglukasik** [gregcman/sucle#23]
Hello! I am a Lisp beginner, and this project looks like a fun way to tinker with an existing codebase.

However, I have the same issue as #9 , where textures do not appear.  `glxinfo | grep -i opengl` confirms I am using OpenGL version 3.2, which no longer supports display lists. (For other OpenGL newbies like me, I found [these online class notes](http://www.cs.uregina.ca/Links/class-info/315/WWW/Lab6/index.html) helpful for understanding the difference between display lists and vertex arrays.)

I can confirm my local development setup (using lem) is working, as I can make some changes to things like the `sky-color` [here](https://github.com/gmasching/sucle/blob/2146b25ed87c9962d2de7a745bd8bb022ff9d7e1/application/testbed/sandbox-subsystem.lisp#L517) and re-evaluate it and see my changes.

My current understanding is that the code change will need to be in `application/subsystems/sandbox/change-world.lisp`, where at the top of the file you find a `draw-world` function that iterates over all the data in the `*g/chunk-call-list*` and uses [`gl:call-list`](https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glCallList.xml) to execute the display list. The equivalent for vertex arrays seems to be something like [`gl:draw-elements`](https://www.khronos.org/registry/OpenGL-Refpages/es3/html/glDrawElements.xhtml), which i see in use in the `draw-vertex-array` function in `src/opengl/glhelp.lisp`.

So it looks like this functionality is (partly) implemented? But I'm not sure how to plug this into the existing code. I'm thinking i'll need to initialize a vao object, and then change `set-chunk-display-list` to update this vao object, and then call `draw-vertex-array` in the `draw-world` function.

I see some code that looks apropos in `dump.lisp` under the 'drawing' heading, but my first attempts at cargo-culting what i saw there were unsuccessful, so i figured I'd try opening an issue as the INSTALL.md suggested. Any pointers would be greatly appreciated!

Thank you for releasing this! I'm hoping i can see it for myself soon :slightly_smiling_face:

**Comment by gmasching** [comment]
@jglukasik 
I'm very, very sorry for the delayed response. 

It seems like you got pretty far trying to understand how everything works, and you were correct. I'm impressed someone was able to decipher how everything worked in this one-off mess that I created. 
However, at the time there was no possible way you could have refactored sucle to use vertex-arrays rather than display lists, because all of the code was disorganized and scattered everywhere. 

So I updated sucle to use either display-lists or vertex-arrays based on the opengl version.
If you want to see how it works, `create-vao-or-display-list-from-specs` in https://github.com/gmasching/sucle/blob/master/src/opengl/opengl-helper.lisp is where to start.
Basically, everything is first drawn to a temporary lisp array. Then at runtime it checks to see if display-lists or vertex-arrays are to be used. 

Thanks for checking out this project, and again sorry for the delayed response.
